File: app/src/main/java/es/etologic/mahjongscoring2/app/main/combinations/CombinationsViewModelFactory.java
package es.etologic.mahjongscoring2.app.main.combinations;

import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.support.annotation.NonNull;

import javax.inject.Inject;

import es.etologic.mahjongscoring2.domain.use_cases.GetCombinationsUseCase;

public class CombinationsViewModelFactory extends ViewModelProvider.NewInstanceFactory {

    private final GetCombinationsUseCase getCombinationsUseCase;

    @Inject
    public CombinationsViewModelFactory(GetCombinationsUseCase getCombinationsUseCase) {
        this.getCombinationsUseCase = getCombinationsUseCase;
    }

    @SuppressWarnings("unchecked")
    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new CombinationsViewModel(getCombinationsUseCase);
    }
}

File: app/src/main/java/es/etologic/mahjongscoring2/app/main/combinations/CombinationsViewModelFactory.kt
package es.etologic.mahjongscoring2.app.main.combinations

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import es.etologic.mahjongscoring2.domain.use_cases.GetCombinationsUseCase
import javax.inject.Inject

class CombinationsViewModelFactory @Inject
constructor(private val getCombinationsUseCase: GetCombinationsUseCase) : ViewModelProvider.NewInstanceFactory() {
    
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        @Suppress("UNCHECKED_CAST")
        return CombinationsViewModel(getCombinationsUseCase) as T
    }
}
